pkg/tfbridge/adapt_check_failures.go

```diff
@@ -91,7 +91,7 @@ func parseCheckError(
 		return pp, failType, s
 	}
 	// If there is no way to identify a propertyPath, still report a generic CheckFailure.
-	return nil, MiscFailure, d.Error()
+	return nil, MiscFailure, err.Error()
```

Comment:
`d` is guaranteed to be `nil` at this point. Could we change the scope of `d` to avoid this kind of mistake?
```
	if d := (*diagnostics.ValidationError)(nil); errors.As(err, &d) {
```

That makes L94 a compile error.

Reply:
Gah, err.Error() is the right thing to do here! Good catch!

I can give more context around this if needed beyond the existing comments. Were trying to parse out diagnostics out of TF and ensure they look better in Pulumi here.